packages/app/src/api/repositories/worldRepository/worldRepository.api.ts

```diff
@@ -23,3 +25,17 @@ export const fetchSpaceWithSubSpaces: RequestInterface<
   const url = generatePath(worldRepositoryEndpoints().getSpaceWithSubspaces, {worldId});
   return request.get(url, restOptions);
 };
+
+export const searchSpaces: RequestInterface<SearchSpacesRequest, SearchSpacesResponse> = (
+  options
+) => {
+  const {worldId, query, ...restOptions} = options;
+
+  restOptions.params = {
+    ...restOptions.params,
+    query: query
```

Comment:
No need for duplication
```suggestion
    query
```